incubator/vault/templates/deployment.yaml

```diff
@@ -49,6 +49,10 @@ spec:
         resources:
 {{ toYaml .Values.resources | indent 10 }}
       affinity:
+        {{- if .Values.nodeAffinity }}
```

Comment:
This is a little inconsistent. I'd prefer it to just change the whole affinity config like so:

```yaml
      {{- if .Values.affinity }}
      affinity:
{{ toYaml .Values.affinity | indent 8 }}
      {{- end }}
```

@jpds

Reply:
@unguiculus Personally I think that would be a good idea, but the current PodAntiAffinity should become part of the default value then I guess. I suppose the original intention of the current PodAntiAffinity rules is to try and place Vault pods on different hosts, which makes sense. But it would be good to be able to customize this as it so you can force pods to be in different AZ's etc.

Reply:
Hmm, I don't think that would work as the podAntiAffinity rules contains templating which means it can't be moved to values.yaml?

Reply:
I thiunk it's ok to change the existing default. As you mention it would be more flexible. Let's see if @jpds jumps in. Otherwise, I'd just go ahead and do it.

Reply:
@unguiculus Alright. But that still leaves the problem that the podAntiAffinity rules can't be moved to values.yaml as they contain Go templating and that can't be parsed inside values.yaml, correct?

Reply:
Yes, users would have to configure that themselves.

Reply:
But how could they get the exact same behaviour as it is now if they wanted it? They would have to hardcode the matchLabels into their values.yaml which seems very counter-intuitive :-/

Reply:
I like @unguiculus's suggestion of keeping this in line with what `helm create` produces by default.